oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/DataModelService.kt

```diff
@@ -1,92 +1,176 @@
 package com.empowerops.volition.ref_oasis
 
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.javafx.JavaFx
-import kotlinx.coroutines.launch
+import com.google.common.eventbus.EventBus
 import java.time.Duration
 
-class DataModelService(
-        val viewData: ViewData,
-        private val updateCallback: () -> Unit
-) : ModelService {
-    override var simByName: Map<String, Simulation> = emptyMap()
+interface Event
+data class StatusUpdateEvent(val status: String) : Event
+data class NewMessageEvent(val message: Message) : Event
+data class NewResultEvent(val result: Result) : Event
+data class PluginRegisteredEvent(val name: String) : Event
+data class PluginUnRegisteredEvent(val name: String) : Event
+class PluginUpdatedEvent : Event
+data class PluginRenamedEvent(val oldName: String, val newName: String) : Event
+data class ProxyAddedEvent(val name: String) : Event
+data class ProxyRemovedEvent(val name: String) : Event
+data class ProxyRenamedEvent(val oldName: String, val newName: String) : Event
+class ProxyUpdatedEvent : Event
+data class IssuesUpdatedEvent(val issues: List<String>) : Event
 
-    override fun updateStatusMessage(message: String) {
-        GlobalScope.launch(Dispatchers.JavaFx) {
-            viewData.currentEvaluationStatus.value = message
-        }
+class DataModelService(private val eventBus: EventBus) {
+    var simByName: Map<String, Simulation> = emptyMap()
+    var proxyByName: Map<String, Proxy> = emptyMap()
+
+    fun updateStatusMessage(message: String) {
+        eventBus.post(StatusUpdateEvent(message))
     }
 
-    override fun addMessage(message: Message) {
-        GlobalScope.launch(Dispatchers.JavaFx) {
-            viewData.allMessages.add(message)
-        }
+    fun addMessage(message: Message) {
+        eventBus.post(NewMessageEvent(message))
     }
```

Comment:
so, dont these strike you as middle men? Can we simply have these callers just use `eventBus.post(newMessageEvent())` instead of deleging to this middle-man?

Reply:
They sure do and I was intend to leave it as it. The problem we had here is we don't have a list of the message in our data model, which is this function was originally intended to do and currently message are view-only data. I think we can just inline these for now.

Reply:
I think eventually we will need to address this and put them back, but I will do that later. i.e. this is not headless ready